core/new-gui/src/app/dashboard/user/component/user-project/user-project.component.ts

```diff
@@ -26,90 +25,18 @@ export class UserProjectComponent implements OnInit {
   public colorBrightnessMap: Map<number, boolean> = new Map(); // tracks brightness of each project's color, to make sure info remains visible against white background
   public colorInputToggleArray: boolean[] = []; // tracks which project's color wheel is toggled on or off
 
-  public readonly ROUTER_USER_PROJECT_BASE_URL = "/dashboard/user-project";
-  public readonly MAX_PROJECT_DESCRIPTION_CHAR_COUNT = 10000;
-
   constructor(
     private userProjectService: UserProjectService,
-    private router: Router,
     private notificationService: NotificationService
   ) {}
 
   ngOnInit(): void {
     this.getUserProjectArray();
   }
 
-  public removeEditNameStatus(pid: number) {
-    this.userProjectEntriesIsEditingName = this.userProjectEntriesIsEditingName.filter(index => index != pid);
-  }
-
-  public removeEditDescriptionStatus(pid: number) {
-    this.userProjectEntriesIsEditingDescription = this.userProjectEntriesIsEditingDescription.filter(
-      index => index != pid
-    );
-  }
-
-  public removeCollapsedProjectDescriptionStatus(pid: number) {
-    this.collapsedProjectDescriptions = this.collapsedProjectDescriptions.filter(index => index != pid);
-  }
-
-  public saveProjectDescription(project: UserProject, newDescr: string, index: number): void {
-    // nothing happens if description is the same
-    if (project.description === newDescr) {
-      this.removeEditDescriptionStatus(project.pid);
-      return;
-    }
-
-    // update the project's description
-    this.userProjectService
-      .updateProjectDescription(project.pid, newDescr)
-      .pipe(untilDestroyed(this))
-      .subscribe(
-        () => {
-          let updatedProjectEntry = { ...project };
-          updatedProjectEntry.description = newDescr;
-          const newEntries = this.userProjectEntries.slice();
-          newEntries[index] = updatedProjectEntry;
-          this.userProjectEntries = newEntries;
-          this.notificationService.success(`Saved description for project: "${project.name}".`);
-        },
-        (err: unknown) => {
-          // @ts-ignore
-          this.notificationService.error(err.error.message);
-        }
-      )
-      .add(() => {
-        this.removeEditDescriptionStatus(project.pid);
-      });
-  }
-
-  public saveProjectName(project: UserProject, newName: string): void {
-    // nothing happens if name is the same
-    if (project.name === newName) {
-      this.removeEditNameStatus(project.pid);
-    } else if (this.isValidNewProjectName(newName, project)) {
```

Comment:
In the copied version of this function in the other file, I removed the `isValidNewProjectName` check because it requires passing the full array into the `user-project-list-item` component. I feel like the added complexity in maintaining this reference isn't worth it.
If there is a duplicate name, the server will return an error and an error message will still be displayed.

Reply:
Makes sense